sqlglot/optimizer/eliminate_join_marks.py

```diff
@@ -0,0 +1,184 @@
+import typing as t
+import copy
+
+from sqlglot import expressions as exp
+from sqlglot.optimizer.scope import build_scope, find_all_in_scope
+
+
+def _update_from(
+    select: exp.Select,
+    new_join_dict: t.Dict[str, exp.Join],
+    old_join_dict: t.Dict[str, exp.Join],
+):
+    """If the from clause needs to become a new join, find an appropriate table to use as the new from.
+    updates select in place
+
+    Args:
+        select (exp.Select): The select statement to update
+        new_join_dict (Dict[str, exp.Join]): The dictionary of new joins
+        old_join_dict (Dict[str, exp.Join]): The dictionary of old joins
+    """
+    old_from = select.args["from"]
+    if old_from.alias_or_name not in new_join_dict.keys():
```

Comment:
remove the calls to keys with in, they are unnecessary

Reply:
removed